website/datafuse/docs/rfcs/query/2021-05-27-data-shuffle.md

```diff
@@ -0,0 +1,138 @@
+# Distributed query and data shuffle
+
+## Summary
+
+Distributed query is distributed database necessary feature.
+This doc is intended to explain the distributed query and its data flow design.
+
+## Local query
+
+Let's see how normal queries run on a single database node.
+
+``` text
+'        +------+       +------------+      +---------+
+'        |      |  AST  |            | Plan |         |
+' SQL--->|Parser+------>|Plan Builder+----->|Optimizer|
+'        |      |       |            |      |         |
+'        +------+       +------------+      +---+-----+
+'                                               | Plan
+'                                               v
+'                 +----------+            +-----------+
+'                 |          |  Processor |           |
+'     Data <------+DataStream|<-----------+Interpreter|
+'                 |          |            |           |
+'                 +----------+            +-----------+
+```
+
+### Parser and AST
+
+DataFuse uses the third-party SQL parser and its AST.
+For more information, see: https://github.com/ballista-compute/sqlparser-rs
+
+### PlanBuilder and Plan
+
+A query plan (or query execution plan) is a sequence of steps used to access data in DataFuse. It is built by PlanBuilder from AST. We also use tree to describe it(similar to AST). But it has some differences with AST:
+
+- Plan is serializable and deserializable.
+- Plan is grammatically safe, we don't worry about it.
```

Comment:
Is this because the plan has passed the parser?

Reply:
yep